test/test_solve_shifted_system.jl

```diff
@@ -0,0 +1,132 @@
+using Test
+using LinearOperators
+using LinearAlgebra
+
+function setup_test_val(; M = 5, n = 100, scaling = false)
+  σ = 0.1
+  B = LBFGSOperator(n, mem = 5, scaling = scaling)
+  s = ones(n)
+  y = ones(n)
+
+  S = randn(n, M)
+  Y = randn(n, M)
+
+  # make sure it is positive
+  for i = 1:M
+    if dot(S[:, i], Y[:, i]) < 0
+      S[:, i] = -S[:, i]
+    end
+  end
+  for i = 1:M
+    s = S[:, i]
+    y = Y[:, i]
+    if dot(s, y) > 1.0e-20
+      push!(B, s, y)
+    end
+  end
```

Comment:
`push!()` already checks the sign. It would be simpler and more efficient to just do
```julia
for _ = 1:10
  s = rand(n)
  y = rand(n)
  push!(B, s, y)
end
```